substrate/validator-sets/pallet/src/lib.rs

```diff
@@ -490,11 +502,12 @@ pub mod pallet {
       network: NetworkId,
       account: T::AccountId,
       amount: Amount,
+      block_reward: bool,
     ) -> DispatchResult {
       let old_allocation = Self::allocation((network, account)).unwrap_or(Amount(0)).0;
       let new_allocation = old_allocation + amount.0;
       let allocation_per_key_share = Self::allocation_per_key_share(network).unwrap().0;
-      if new_allocation < allocation_per_key_share {
+      if new_allocation < allocation_per_key_share && !block_reward {
         Err(Error::<T>::InsufficientAllocation)?;
```

Comment:
won't genesis validators have any existing allocations when they start to receive block rewards? Deallocations of block rewards will go through regular deallocation process except for the last session? Last session reward will immediately be deallocatable since the validator won't be in the set.

Reply:
So we do need block rewards -> allocations, and we just need to double check we don't assume present in allocations map == eligible. ACK.